### DISCUSSION TITLE
Can I map a key press to `next_tab`, but if there is only one tab, to send the keypress to the

### DISCUSSION THREAD

**Discussion by amandasaurus** [kovidgoyal/kitty#4883] [Q&A]
I have mapped `Ctrl`-`PageDown` in kitty to next tab (`map ctrl+page_down next_tab`). However if I have only one kitty tab, running neovim with >1 neovim tab, and I press `Ctrl`-`PageDown`, nothing happens. I would like neovim to get the keypress and switch to the next neovim tab. I presume because I told kitty to “switch to the next (kitty) tab” and since there is only one kitty tab, nothing happens. Is it possible to configure kitty to send the ctrl-pageup keypress to the running programme if there is only 1 kitty tab open?

I used to use gnome terminal, and have `Ctrl`-`PageUp`/`Down` mapped to Next/Previous Tab, just like kitty. In my (neo)vim config (which also has it's own tabs), I have the same key mapping. With one gnome terminal window open, with one gnome terminal tab open running neovim with >1 neovim tabs, if I press ctrl-pageup, then neovim will switch to next neovim tab. If I have >1 gnome terminal tab, then gnome terminal consumes the keypress and switch to the next gnome-terminal tab. I would like to reproduce this behaviour in kitty.

**Comment by page-down** [comment]
You can write kitten to send keystrokes based on conditions.
First check the number of tabs, then send custom keys.

https://sw.kovidgoyal.net/kitty/kittens/custom/

You can see the examples in the above page.
https://github.com/yurikhan/kitty-smart-scroll

I have guided how to check the number of tabs in the following.
https://github.com/kovidgoyal/kitty/discussions/4363
